bin/hook-reference/index.js

```diff
@@ -0,0 +1,28 @@
+const fs = require( 'fs' );
+const { stat, readdir, writeFile } = require( 'fs/promises' );
+const { resolve } = require( 'path' );
+const createData = require( './data' );
+
+async function getFilePaths( dir ) {
```

Comment:
`Promise.all`, `map`, `reduce`, and _recursion_ all in a ~10 line async function. 👏🏻